Automate - Quota - fixed issue with memory and storage sizes.

billfitzgerald0120 opened the pull request:
https://bugzilla.redhat.com/show_bug.cgi?id=1289278

billfitzgerald0120 commented:
@mkanoor 
Please review

billfitzgerald0120 commented:
@gmcculloug 
Please review

mkanoor commented on db/fixtures/ae_datastore/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb:
```diff
@@ -110,16 +110,12 @@ def requested_number_of_cpus(args_hash)
 
 def requested_storage(args_hash)
-  vm_size = bytes_to_megabytes(args_hash[:resource].vm_template.provisioned_storage)
+  vm_size = args_hash[:resource].vm_template.provisioned_storage
```
@billfitzgerald0120 
Is the provisioned_storage going to be in bytes for all platforms.

mkanoor commented:
@billfitzgerald0120 
The bugzilla ticket talks about changing the log message, is that going to be addressed in this PR
After:
Request exceeds maximum allowed for the following: (cpu - Used: 1719 plus requested: 4 exceeds quota: 1) (memory - Used: 3.27 MB plus requested: 15.00 GB exceeds quota: 2.00 GB) (storage - Used: 4.79 MB plus requested: 80.00 GB exceeds quota:

billfitzgerald0120 commented:
@mkanoor 
The log message now has accurate information and all memory and storage values are expressed as GB.  I have provided a sample below:
(memory - Used: 3145.95 GB plus requested: 2.00 GB exceeds quota: 100.00 GB) (storage - Used: 14809.10 GB plus requested: 18.00 GB exceeds quota: 100.00 GB)

billfitzgerald0120 commented:
@mkanoor 
Correction: Storage values will be expressed in the proper increment.  The problem was that some values were wrong.  The Used values were supposed to be GB and we were expressing them as MB.
Also the requested memory was in KB not bytes so I needed to changed this line to set it to bytes ...
{:memory  => get_total_requested(options_hash, :vm_memory) * 1024**2,

mkanoor commented:
:+1:

billfitzgerald0120 commented:
@mkanoor 
The code was expecting the requested memory to be in bytes but it was in KB.  I modified the 'requested' method to change KB into bytes ({:memory  => get_total_requested(options_hash, :vm_memory) * 1024**2).  The 'bytes_to_megabytes' routine in the 'used' method was also causing problems so I removed it.  The original BZ:  https://bugzilla.redhat.com/show_bug.cgi?id=1289278 mentions that the values were incorrect.  That is the log formatting problem.
I also modified the 2 rspecs to handle the proper values.

billfitzgerald0120 commented:
@gmcculloug 
Please review
